### PR TITLE
Feature/dsg 2805/invitation by

### PR DESCRIPTION
**Pull request by jcsuarezm**
## Types of changes

Patch
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation change (documentation only, will not require redeployment)
- [ ] Refactor or cleanup (functionality unchanged)
Minor
- [ ] New feature (non-breaking change which adds functionality)
Major
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

- [ ] I have confirmed all acceptance criteria defined by the story have been met

### CODE REVIEW

File: service/src/main/java/io/nuvalence/workmanager/service/controllers/IndividualProfilesApiDelegateImpl.java [hunk 1]
```diff
@@ -517,6 +517,34 @@ public ResponseEntity<Void> deleteIndividualProfileInvitation(
...
+        if (!authorizationHandler.isAllowed("read", INDIVIDUAL_PROFILE_INVITATION_CERBOS_TYPE)) {
+            throw new ForbiddenException();
+        }
+
+        ProfileInvitation invitation =
+                profileInvitationService
+                        .getInvitationByIdAndType(invitationId, ProfileType.INDIVIDUAL)
+                        .orElseThrow(() -> new NotFoundException("Invitation not found"));
+
+        if (UserUtility.getAuthenticatedUserType().equalsIgnoreCase(UserType.PUBLIC.getValue())) {
```
Comment (kives-nu) [inline service/src/main/java/io/nuvalence/workmanager/service/controllers/IndividualProfilesApiDelegateImpl.java:532]: Just to confirm, this auth flow will let reader/writer/admin of a profile view the invite ? That makes sense to me but want to confirm thats the desired functionality

File: service/src/test/java/io/nuvalence/workmanager/service/controllers/IndividualProfilesApiDelegateImplTest.java [hunk 2]
```diff
@@ -1017,6 +1019,119 @@ void deleteIndividualProfileInvitation_forbidden() throws Exception {
...
+                    .andExpect(
+                            content()
+                                    .string(
+                                            containsString(
+                                                    "User does not have access to this profile")));
+        }
+    }
+
+    @Test
+    void getIndividualProfileInvitationById_success() throws Exception {
```
Comment (kives-nu) [inline service/src/test/java/io/nuvalence/workmanager/service/controllers/IndividualProfilesApiDelegateImplTest.java:1085]: Nice (y)

Comment (kives-nu) [reply service/src/test/java/io/nuvalence/workmanager/service/controllers/IndividualProfilesApiDelegateImplTest.java:1085]: Should we maybe have a test that confirms public users with a profile link can also get the invites ? I think this is just checking the flow of public users with a matching email right ?

**Review by kives-nu — APPROVED** [review]
Left two small comments for clarification but looks good to me

File: service/src/test/java/io/nuvalence/workmanager/service/controllers/IndividualProfilesApiDelegateImplTest.java (hunk 2 above)
Comment (jcsuarezm) [reply service/src/test/java/io/nuvalence/workmanager/service/controllers/IndividualProfilesApiDelegateImplTest.java:1085]: Hi Kenneth, thank you! I can split this into two tests to cover them separately as in this test the user is authorized both by links and email. I will do it in a follow-up PR to unblock work earlier.

File: service/src/main/java/io/nuvalence/workmanager/service/controllers/IndividualProfilesApiDelegateImpl.java (hunk 1 above)
Comment (jcsuarezm) [reply service/src/main/java/io/nuvalence/workmanager/service/controllers/IndividualProfilesApiDelegateImpl.java:532]: Hi Kenneth, thank you, that is correct, I am not validating access level as all levels can read.

**Pull request merged**